cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_group_facts
+short_description: Gather facts about security groups in AWS
+description:
+    - Gather facts about security groups in AWS
+version_added: "2.2"
+author: "Destry Jaimes (@calamityman)"
+options:
+  filters:
```

Comment:
Those options are actually pretty confusing and `filters` already takes care of any selection needs that you might have. I recommend keeping group names and group ids options out of this module.